nautobot_floor_plan/static/nautobot_floor_plan/js/floorplan.js

```diff
@@ -731,4 +719,58 @@ document.addEventListener('DOMContentLoaded', function() {
 
     // Call the function when the page loads
     highlightElementFromURL();
+
+    // Add function to check the current theme
+    function getCurrentTheme() {
+        // First check localStorage for user preference
+        const currentTheme = localStorage.getItem('theme');
+        if (currentTheme && currentTheme !== "system") {
+            return currentTheme;
+        }
+
+        // If system theme, check system preference
+        if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+            return "dark";
+        }
+
+        // Default to light
+        return "light";
+    }
+
+    // Add this function to update the SVG's CSS
+    function updateSvgTheme() {
+        const isDarkMode = getCurrentTheme() === "dark";
+        const svgElement = document.querySelector('#floor-plan-svg');
+        if (svgElement) {
+            // Find the style element within the SVG
+            const styleElement = svgElement.querySelector('style');
+            if (styleElement) {
+                // Fetch the appropriate CSS file
+                const cssFile = isDarkMode ? 'dark_svg.css' : 'svg.css';
+                fetch(`/static/nautobot_floor_plan/css/${cssFile}`)
+                    .then(response => response.text())
+                    .then(css => {
+                        styleElement.textContent = css;
+                });
+            }
+        }
+    }
+
+    // Call this when the page loads
+    updateSvgTheme();
+
+    // Also update when theme changes
+    const htmlEl = document.getElementsByTagName('html')[0];
+    const observer = new MutationObserver(function(mutations) {
+        mutations.forEach(function(mutation) {
+            if (mutation.attributeName === 'data-theme') {
+                updateSvgTheme();
+            }
+        });
+    });
+
+    observer.observe(htmlEl, {
+        attributes: true,
+        attributeFilter: ['data-theme']
+    });
```

Comment:
Just for my own edification, can you add some comments to explain how this works?

Reply:
Awesome